plugins/catalog/src/components/CatalogPage/utils/timeUtil.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import moment from 'moment';
+
+/**
+ * Validates that a date is a valid ISO string.
+ *
+ * @param date A date string
+ * @returns {bool} Whether the date is valid or not.
+ */
+export function isValidDate(date) {
+  return moment(date).isValid();
+}
+
+/**
+ * Validates that a date is a valid ISO string and a specific format.
+ *
+ * @param date A date string
+ * @param format A format string or an array of format strings to validate against.
+ * @returns {bool} Whether the date is valid or not according to the format.
+ */
+export function isValidDateAndFormat(date, format) {
+  return moment(date, format, true).isValid();
+}
+
+export function relativeTime(timestamp) {
+  return moment(timestamp).fromNow();
+}
+
+// Select a large random integer at startup, to prevent the greetings to change every time the user
+// navigates.
+const greetingRandomSeed = Math.floor(Math.random() * 1000000);
+
+export function getTimeBasedGreeting() {
+  const random = array => array[greetingRandomSeed % array.length];
+
+  const currentHour = new Date(Date.now()).getHours();
+  if (currentHour >= 23) {
+    return {
+      language: 'Seriously',
+      greeting: 'Get some rest',
+    };
+  }
+  const timeOfDay = hour => {
+    if (hour < 12) return 'goodMorning';
+    if (hour < 17) return 'goodAfternoon';
+    return 'goodEvening';
+  };
+  // TODO: This does not work.
+  const greetings = require(`utils/locales/${timeOfDay(currentHour)}.locales.json`);
```

Comment:
Can someone explain why this does not work? Some webpack magic that I don't understand 🤔

Reply:
It's actually lack of magic 😅. We removed src-relative imports so you just have to give a relative path instead. I don't think this dynamic require helps much though tbh, either just import all 3 and select the one we want, or use dynamic imports (´import()´) if we wanna save on the bundle size.

Reply:
@Rugvip are you fine this fix?

Reply:
Yup, could make a good first issue about moving it to dynamic imports if you're in the mood for that :grin:

Reply:
For sure! Will create one when this lands.